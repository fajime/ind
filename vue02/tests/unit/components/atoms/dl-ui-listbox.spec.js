import { mount } from '@vue/test-utils';
import DlUiListbox from '../../../../src/components/atoms/dl-ui-listbox/index.vue';

describe('dl-ui-listbox', () => {

  const wrapper = mount(DlUiListbox);

  it('Component Name', () => expect(DlUiListbox.name).toBe('dl-ui-listbox'));

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('has a container', () => {
    expect(wrapper.vm.$refs.container).not.toBeUndefined();
  });

  it('has listbox', () => {
    expect(wrapper.find({ ref : 'list' }));
  });

  it('has an option', () => {
    expect(wrapper.find({ ref : 'option' }));
  });

  it('has a checkbox', async () => {
    wrapper.setProps({ multiple : true });
    await wrapper.vm.$nextTick();
    expect(wrapper.find('dl-ui-checkbox')).toBeTruthy();

  });

  it('check icons', async () => {
    const wrapperTest = mount(DlUiListbox, {
      propsData : {
        options : [
          { name : 'Australia', value : 'AU', leftIcon : '' },
          { name : 'Brazil', value : 'BR' }
        ]
      }
    });
    wrapperTest.setProps({ leftIcon : 'dl-ui-icon-user' });
    wrapperTest.setProps({ rightIcon : 'dl-ui-icon-lab' });
    await wrapperTest.vm.$nextTick();
    expect(wrapper.find('dl-ui-listbox__option-left-icon')).toBeTruthy();
    expect(wrapper.find('dl-ui-listbox__option-right-icon')).toBeTruthy();
    wrapperTest.setProps({ leftIcon : undefined });
    wrapperTest.setProps({ rightIcon : undefined });
    await wrapperTest.vm.$nextTick();
    expect(wrapperTest.vm.$children[0]).toBeUndefined();
    expect(wrapperTest.vm.$children[1]).toBeUndefined();
  });

  describe('methods', () => {
    it('isSelected', () => {
      const wrapperTest = mount(DlUiListbox, {
        propsData : {
          options : [
            { name : 'Australia', value : 'AU', leftIcon : '' },
            { name : 'Brazil', value : 'BR' }
          ],
          value    : ['AU'],
          multiple : true
        }
      });
      const firstOption = wrapperTest.vm.$options.propsData.options[0];
      expect(wrapperTest.vm.isSelected(firstOption)).toBeTruthy();
    });

    it('onOptionSelect', async () => {
      const event = { target : { value : 'fake ' } };
      const wrapperTest = mount(DlUiListbox, {
        propsData : {
          options : [
            { name : 'Australia', value : 'AU', leftIcon : '' },
            { name : 'Brazil', value : 'BR' }
          ],
          multiple : true
        }
      });
      const firstOption = wrapperTest.vm.$options.propsData.options[0];
      wrapperTest.vm.onOptionSelect(event, firstOption);
      wrapperTest.setProps({ value : ['AU'] });
      await wrapperTest.vm.$nextTick();
      expect(wrapperTest.emitted('input')).toBeTruthy();
      wrapperTest.vm.onOptionSelect(event, firstOption);
      wrapperTest.setProps({ multiple : false });
      await wrapperTest.vm.$nextTick();
      wrapperTest.vm.onOptionSelect(event, firstOption);
    });
  });

  describe('computed', () => {

    it('scrollHeight - not mounted', async () => {
      const wrapperTest = mount(DlUiListbox);
      wrapperTest.setData({ isMounted : false });
      await wrapperTest.vm.$nextTick();
      expect(wrapperTest.vm.scrollHeight).toBe('');
    });

    it('scrollHeight - is mounted && no elements', async () => {
      const fn = window.getComputedStyle;
      const stub = jest.fn(() => ({ getPropertyValue : () => 100 }));
      window.getComputedStyle = stub;
      const wrapperTest = mount(DlUiListbox);
      wrapperTest.setData({ isMounted : true });
      await wrapperTest.vm.$nextTick();
      expect(wrapperTest.vm.scrollHeight).toBe('360px');
      window.getComputedStyle = fn;
    });

  });
});
