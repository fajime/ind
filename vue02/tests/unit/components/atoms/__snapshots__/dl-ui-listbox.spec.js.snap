// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dl-ui-listbox renders correctly 1`] = `
<div
  class="dl-ui-listbox"
>
  <div
    class="dl-ui-listbox__container"
  >
    <div
      class="dl-ui-scrollpanel dl-ui-scrollbar"
      style="max-height: auto;"
    >
      
      <div
        class="dl-ui-list__items-wrapper"
      >
        <ul
          class="dl-ui-list__items"
        >
          
          <li
            class="dl-ui-list__item"
            tabindex="0"
          >
            
            <div
              class="dl-ui-rowitem dl-ui-rowitem__option dl-ui-rowitem"
            >
              <!--v-if-->
              <!--v-if-->
              <div
                class="dl-ui-rowitem__box-text"
              >
                name 1
              </div>
              <!--v-if-->
            </div>
            
          </li>
          <li
            class="dl-ui-list__item"
            tabindex="0"
          >
            
            <div
              class="dl-ui-rowitem dl-ui-rowitem__option dl-ui-rowitem"
            >
              <!--v-if-->
              <div
                class="dl-ui-icon dl-ui-rowitem__option-left-icon"
              >
                <svg>
                  <!--v-if-->
                  <!--v-if-->
                  <use
                    xlink:href="#dl-ids-icon-user-single-outlined"
                  />
                </svg>
              </div>
              <div
                class="dl-ui-rowitem__box-text"
              >
                name 2
              </div>
              <!--v-if-->
            </div>
            
          </li>
          <li
            class="dl-ui-list__item"
            tabindex="0"
          >
            
            <div
              class="dl-ui-rowitem dl-ui-rowitem__option dl-ui-rowitem"
            >
              <!--v-if-->
              <!--v-if-->
              <div
                class="dl-ui-rowitem__box-text"
              >
                name 3
              </div>
              <div
                class="dl-ui-icon dl-ui-rowitem__option-right-icon"
              >
                <svg>
                  <!--v-if-->
                  <!--v-if-->
                  <use
                    xlink:href="#dl-ids-icon-user-single-outlined"
                  />
                </svg>
              </div>
            </div>
            
          </li>
          <li
            class="dl-ui-list__item dl-ui-list__item--disabled"
          >
            
            <div
              class="dl-ui-rowitem dl-ui-rowitem__option dl-ui-rowitem"
            >
              <!--v-if-->
              <!--v-if-->
              <div
                class="dl-ui-rowitem__box-text"
              >
                name 4
              </div>
              <!--v-if-->
            </div>
            
          </li>
          <li
            class="dl-ui-list__item"
            tabindex="0"
          >
            
            <div
              class="dl-ui-rowitem dl-ui-rowitem__option dl-ui-rowitem"
            >
              <!--v-if-->
              <!--v-if-->
              <div
                class="dl-ui-rowitem__box-text"
              >
                name 5
              </div>
              <!--v-if-->
            </div>
            
          </li>
          
        </ul>
      </div>
      
    </div>
  </div>
</div>
`;
